gs/backend/positioning/spg4.py

```
@@ -0,0 +1,50 @@
+import sgp4 as _sgp4
+import sgp4.api as _sgp4API
+from datetime import datetime
+from sgp4.api import Satrec, jday, WGS72, SGP4_ERRORS
+from backend.positioning.tle import TLEData
+import dataclasses
+from typing import Optional
+
+
+@dataclasses.dataclass
+class SGP4Data:
+   position_km: tuple[float, float, float] 
+   velocity_km_sec: tuple[float, float, float]
+   error : Optional[str] = None #should I set this as optional to default as none or is the redundancy not needed
```

You should raise an error when there is an error instead of having a field that is a part of dataclass